cmd/difference.go

```diff
@@ -265,7 +267,7 @@ func differenceInternal(ctx context.Context, sourceClnt, targetClnt Client, sour
 				}
 				continue
 			}
-			if (srcType.IsRegular() && tgtType.IsRegular()) && srcSize != tgtSize {
+			if srcSize != tgtSize {
```

Comment:
If mtime for the source object is older than the destinaton object's mtime, but their sizes are different, then we'll copy over the newer version of the file just because sizes are different.

```
$ mc stat tmp1/content.txt
Name      : content.txt
Date      : 2020-09-11 19:41:15 PDT 
Size      : 30 B   
Type      : file 
Metadata  :
  Content-Type       : text/plain 
  X-Amz-Meta-Mc-Attrs: atime:1599878236#794570223/gid:1000/gname:ersan/mode:33188/mtime:1599878475#869739459/uid:1000/uname:ersan 

$ mc stat tmp/content.txt
Name      : content.txt
Date      : 2020-09-11 03:06:55 PDT 
Size      : 28 B   
Type      : file 
Metadata  :
  X-Amz-Meta-Mc-Attrs: atime:1599820232#311739372/gid:1000/gname:ersan/mode:33188/mtime:1599818815#52759230/uid:1000/uname:ersan 
  Content-Type       : text/plain 

$ mc mirror -a tmp1/ myminio/dst/
...minio/mc/tmp1/content.txt:  30 B / 30 B ┃▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓┃ 2.36 KiB/s 0s
$ mc mirror -a tmp/ myminio/dst/
.../minio/mc/tmp/content.txt:  28 B / 28 B ┃▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓┃ 3.93 KiB/s 0s
```

I don't think we need to check the size anymore since we've mtime check going forward.
We can safely remove it.

Reply:
That is okay it is not the intention @ebozduman of this PR to remove older checks.